crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -84,11 +158,13 @@ where
         l2_block_source: L2,
         metadata_storage: Metadata,
     ) -> Result<Self> {
+        let target_block_height = init_metadata.l2_block_height();
         let inner = metadata_storage
-            .get_metadata()
+            .get_metadata(&target_block_height)
             .await?
```

Comment:
The same here, I think we don't need to require usage of `async` it can be sync.